qcodes/instrument_drivers/AlazarTech/ATS9440.py

```diff
@@ -0,0 +1,311 @@
+                               unit='V',
+                               initial_value=0.4,
+                               val_mapping={0.4: 7})
+            self.add_parameter(name='impedance' + i,
+                               get_cmd=None,
+                               parameter_class=TraceParameter,
+                               label='Impedance channel ' + i,
+                               unit='Ohm',
+                               initial_value=50,
+                               val_mapping={50: 2})
+
+            self.add_parameter(name='bwlimit' + i,
+                               get_cmd=None,
+                               parameter_class=TraceParameter,
+                               label='Bandwidth limit channel ' + i,
+                               unit=None,
+                               initial_value='DISABLED',
+                               val_mapping={'DISABLED': 0,
+                                            'ENABLED': 1})
+        self.add_parameter(name='trigger_operation',
+                           get_cmd=None,
+                           parameter_class=TraceParameter,
+                           label='Trigger Operation',
+                           unit=None,
+                           initial_value='TRIG_ENGINE_OP_J',
+                           val_mapping={'TRIG_ENGINE_OP_J': 0,
+                                        'TRIG_ENGINE_OP_K': 1,
+                                        'TRIG_ENGINE_OP_J_OR_K': 2,
+                                        'TRIG_ENGINE_OP_J_AND_K': 3,
+                                        'TRIG_ENGINE_OP_J_XOR_K': 4,
+                                        'TRIG_ENGINE_OP_J_AND_NOT_K': 5,
+                                        'TRIG_ENGINE_OP_NOT_J_AND_K': 6})
+        for i in ['1', '2']:
+            self.add_parameter(name='trigger_engine' + i,
+                               get_cmd=None,
+                               parameter_class=TraceParameter,
+                               label='Trigger Engine ' + i,
+                               unit=None,
+                               initial_value='TRIG_ENGINE_' + ('J' if i == '1' else 'K'),
+                               val_mapping={'TRIG_ENGINE_J': 0,
+                                            'TRIG_ENGINE_K': 1})
+            self.add_parameter(name='trigger_source' + i,
+                               get_cmd=None,
+                               parameter_class=TraceParameter,
+                               label='Trigger Source ' + i,
+                               unit=None,
+                               initial_value='EXTERNAL',
+                               val_mapping={'CHANNEL_A': 0,
+                                            'CHANNEL_B': 1,
+                                            'EXTERNAL': 2,
+                                            'DISABLE': 3,
+                                            'CHANNEL_C': 4,
+                                            'CHANNEL_D': 5})
```

Comment:
Maybe this is a bit cleaner:
val_mapping={'CHANNEL_A': 0,
                                            'CHANNEL_B': 1,
                                            'CHANNEL_C': 2,
                                            'CHANNEL_D': 3,
                                            'EXTERNAL': 4,
                                            'DISABLE': 5,})